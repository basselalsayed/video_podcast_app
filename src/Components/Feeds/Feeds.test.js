import React from 'React';
import { Feed, Feeds } from '../';

const feeds = [...Array(4).keys()];
const defaultProps = { feeds };

describe('<Feeds />', () => {
  const wrapper = shallow(<Feeds {...defaultProps} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has 4 Feeds', () => {
    expect(wrapper.find(Feed).length).toEqual(4);
  });
});
