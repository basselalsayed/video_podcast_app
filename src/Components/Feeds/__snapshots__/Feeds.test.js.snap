// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feeds /> renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(FeedsBase)>
    <FeedsBase
      currentFeeds={
        Array [
          Object {
            "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
            "pubDate": 3,
            "title": "test 1",
          },
          Object {
            "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
            "pubDate": 3,
            "title": "test 2",
          },
          Object {
            "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
            "pubDate": 3,
            "title": "test 3",
          },
          Object {
            "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
            "pubDate": 3,
            "title": "test 4",
          },
        ]
      }
      disabled={
        Object {
          "next": false,
          "prev": true,
        }
      }
      nextFeedListSection={[Function]}
      prevFeedListSection={[Function]}
    >
      <div
        className="column"
      >
        <Button
          active={false}
          disabled={true}
          id="scroll-prev"
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            id="scroll-prev"
            onClick={[Function]}
            type="button"
          >
            â–²
          </button>
        </Button>
        <ListGroup
          horizontal={null}
          variant={null}
        >
          <ForwardRef
            as="div"
            className="list-group"
            onSelect={[Function]}
          >
            <div
              className="list-group"
              onKeyDown={[Function]}
            >
              <Connect(FeedBase)
                feed={
                  Object {
                    "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                    "pubDate": 3,
                    "title": "test 1",
                  }
                }
                idx={0}
                key="0"
              >
                <FeedBase
                  feed={
                    Object {
                      "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                      "pubDate": 3,
                      "title": "test 1",
                    }
                  }
                  idx={0}
                  setNowPlayingRedux={[Function]}
                >
                  <ListGroupItem
                    action={true}
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                    variant={null}
                  >
                    <ForwardRef
                      as="button"
                      className="list-group-item list-group-item-action"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "padding": 0,
                        }
                      }
                    >
                      <button
                        className="list-group-item list-group-item-action"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                      >
                        <Card
                          body={false}
                        >
                          <div
                            className="card"
                          >
                            <div
                              className="feed"
                            >
                              <VideoThumbnail
                                badge="undefined"
                                classname="vidThumb"
                                message=""
                                muted={true}
                                preview="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                title={null}
                                width="115%"
                              >
                                <div
                                  className="styles_videoThumbnail__1A0YX vidThumb"
                                  style={
                                    Object {
                                      "width": "115%",
                                    }
                                  }
                                >
                                  <div
                                    className="styles_video__fdeKk"
                                    style={
                                      Object {
                                        "width": "115%",
                                      }
                                    }
                                  >
                                    <video
                                      className="styles_preview__2cXKK"
                                      loop={true}
                                      muted={true}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      src="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                      width="115%"
                                    />
                                    <div
                                      className="styles_badge__3dxZ6"
                                      style={
                                        Object {
                                          "background": undefined,
                                        }
                                      }
                                    >
                                      undefined
                                    </div>
                                  </div>
                                  <h4
                                    className="styles_title__1cca0"
                                  />
                                  <span
                                    className="styles_views__2l1fl"
                                  />
                                </div>
                              </VideoThumbnail>
                              <div
                                className="textContent"
                              >
                                <CardTitle>
                                  <ForwardRef
                                    className="card-title"
                                  >
                                    <div
                                      className="card-title h5"
                                    >
                                      test 1
                                    </div>
                                  </ForwardRef>
                                </CardTitle>
                                <CardFooter>
                                  <div
                                    className="card-footer"
                                  >
                                    3
                                  </div>
                                </CardFooter>
                              </div>
                            </div>
                          </div>
                        </Card>
                      </button>
                    </ForwardRef>
                  </ListGroupItem>
                </FeedBase>
              </Connect(FeedBase)>
              <Connect(FeedBase)
                feed={
                  Object {
                    "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                    "pubDate": 3,
                    "title": "test 2",
                  }
                }
                idx={1}
                key="1"
              >
                <FeedBase
                  feed={
                    Object {
                      "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                      "pubDate": 3,
                      "title": "test 2",
                    }
                  }
                  idx={1}
                  setNowPlayingRedux={[Function]}
                >
                  <ListGroupItem
                    action={true}
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                    variant={null}
                  >
                    <ForwardRef
                      as="button"
                      className="list-group-item list-group-item-action"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "padding": 0,
                        }
                      }
                    >
                      <button
                        className="list-group-item list-group-item-action"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                      >
                        <Card
                          body={false}
                        >
                          <div
                            className="card"
                          >
                            <div
                              className="feed"
                            >
                              <VideoThumbnail
                                badge="undefined"
                                classname="vidThumb"
                                message=""
                                muted={true}
                                preview="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                title={null}
                                width="115%"
                              >
                                <div
                                  className="styles_videoThumbnail__1A0YX vidThumb"
                                  style={
                                    Object {
                                      "width": "115%",
                                    }
                                  }
                                >
                                  <div
                                    className="styles_video__fdeKk"
                                    style={
                                      Object {
                                        "width": "115%",
                                      }
                                    }
                                  >
                                    <video
                                      className="styles_preview__2cXKK"
                                      loop={true}
                                      muted={true}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      src="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                      width="115%"
                                    />
                                    <div
                                      className="styles_badge__3dxZ6"
                                      style={
                                        Object {
                                          "background": undefined,
                                        }
                                      }
                                    >
                                      undefined
                                    </div>
                                  </div>
                                  <h4
                                    className="styles_title__1cca0"
                                  />
                                  <span
                                    className="styles_views__2l1fl"
                                  />
                                </div>
                              </VideoThumbnail>
                              <div
                                className="textContent"
                              >
                                <CardTitle>
                                  <ForwardRef
                                    className="card-title"
                                  >
                                    <div
                                      className="card-title h5"
                                    >
                                      test 2
                                    </div>
                                  </ForwardRef>
                                </CardTitle>
                                <CardFooter>
                                  <div
                                    className="card-footer"
                                  >
                                    3
                                  </div>
                                </CardFooter>
                              </div>
                            </div>
                          </div>
                        </Card>
                      </button>
                    </ForwardRef>
                  </ListGroupItem>
                </FeedBase>
              </Connect(FeedBase)>
              <Connect(FeedBase)
                feed={
                  Object {
                    "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                    "pubDate": 3,
                    "title": "test 3",
                  }
                }
                idx={2}
                key="2"
              >
                <FeedBase
                  feed={
                    Object {
                      "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                      "pubDate": 3,
                      "title": "test 3",
                    }
                  }
                  idx={2}
                  setNowPlayingRedux={[Function]}
                >
                  <ListGroupItem
                    action={true}
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                    variant={null}
                  >
                    <ForwardRef
                      as="button"
                      className="list-group-item list-group-item-action"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "padding": 0,
                        }
                      }
                    >
                      <button
                        className="list-group-item list-group-item-action"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                      >
                        <Card
                          body={false}
                        >
                          <div
                            className="card"
                          >
                            <div
                              className="feed"
                            >
                              <VideoThumbnail
                                badge="undefined"
                                classname="vidThumb"
                                message=""
                                muted={true}
                                preview="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                title={null}
                                width="115%"
                              >
                                <div
                                  className="styles_videoThumbnail__1A0YX vidThumb"
                                  style={
                                    Object {
                                      "width": "115%",
                                    }
                                  }
                                >
                                  <div
                                    className="styles_video__fdeKk"
                                    style={
                                      Object {
                                        "width": "115%",
                                      }
                                    }
                                  >
                                    <video
                                      className="styles_preview__2cXKK"
                                      loop={true}
                                      muted={true}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      src="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                      width="115%"
                                    />
                                    <div
                                      className="styles_badge__3dxZ6"
                                      style={
                                        Object {
                                          "background": undefined,
                                        }
                                      }
                                    >
                                      undefined
                                    </div>
                                  </div>
                                  <h4
                                    className="styles_title__1cca0"
                                  />
                                  <span
                                    className="styles_views__2l1fl"
                                  />
                                </div>
                              </VideoThumbnail>
                              <div
                                className="textContent"
                              >
                                <CardTitle>
                                  <ForwardRef
                                    className="card-title"
                                  >
                                    <div
                                      className="card-title h5"
                                    >
                                      test 3
                                    </div>
                                  </ForwardRef>
                                </CardTitle>
                                <CardFooter>
                                  <div
                                    className="card-footer"
                                  >
                                    3
                                  </div>
                                </CardFooter>
                              </div>
                            </div>
                          </div>
                        </Card>
                      </button>
                    </ForwardRef>
                  </ListGroupItem>
                </FeedBase>
              </Connect(FeedBase)>
              <Connect(FeedBase)
                feed={
                  Object {
                    "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                    "pubDate": 3,
                    "title": "test 4",
                  }
                }
                idx={3}
                key="3"
              >
                <FeedBase
                  feed={
                    Object {
                      "link": "https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4",
                      "pubDate": 3,
                      "title": "test 4",
                    }
                  }
                  idx={3}
                  setNowPlayingRedux={[Function]}
                >
                  <ListGroupItem
                    action={true}
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                    variant={null}
                  >
                    <ForwardRef
                      as="button"
                      className="list-group-item list-group-item-action"
                      disabled={false}
                      eventKey={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "padding": 0,
                        }
                      }
                    >
                      <button
                        className="list-group-item list-group-item-action"
                        data-rb-event-key={null}
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                      >
                        <Card
                          body={false}
                        >
                          <div
                            className="card"
                          >
                            <div
                              className="feed"
                            >
                              <VideoThumbnail
                                badge="undefined"
                                classname="vidThumb"
                                message=""
                                muted={true}
                                preview="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                title={null}
                                width="115%"
                              >
                                <div
                                  className="styles_videoThumbnail__1A0YX vidThumb"
                                  style={
                                    Object {
                                      "width": "115%",
                                    }
                                  }
                                >
                                  <div
                                    className="styles_video__fdeKk"
                                    style={
                                      Object {
                                        "width": "115%",
                                      }
                                    }
                                  >
                                    <video
                                      className="styles_preview__2cXKK"
                                      loop={true}
                                      muted={true}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      src="https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4"
                                      width="115%"
                                    />
                                    <div
                                      className="styles_badge__3dxZ6"
                                      style={
                                        Object {
                                          "background": undefined,
                                        }
                                      }
                                    >
                                      undefined
                                    </div>
                                  </div>
                                  <h4
                                    className="styles_title__1cca0"
                                  />
                                  <span
                                    className="styles_views__2l1fl"
                                  />
                                </div>
                              </VideoThumbnail>
                              <div
                                className="textContent"
                              >
                                <CardTitle>
                                  <ForwardRef
                                    className="card-title"
                                  >
                                    <div
                                      className="card-title h5"
                                    >
                                      test 4
                                    </div>
                                  </ForwardRef>
                                </CardTitle>
                                <CardFooter>
                                  <div
                                    className="card-footer"
                                  >
                                    3
                                  </div>
                                </CardFooter>
                              </div>
                            </div>
                          </div>
                        </Card>
                      </button>
                    </ForwardRef>
                  </ListGroupItem>
                </FeedBase>
              </Connect(FeedBase)>
            </div>
          </ForwardRef>
        </ListGroup>
        <Button
          active={false}
          disabled={false}
          id="scroll-next"
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            id="scroll-next"
            onClick={[Function]}
            type="button"
          >
            â–¼
          </button>
        </Button>
      </div>
    </FeedsBase>
  </Connect(FeedsBase)>
</Provider>
`;
